test failures on Windows 7 OS

OS: Windows 7, 64 bit

```
$ python
Python 2.7.8 |Anaconda 2.1.0 (32-bit)| (default, Jul  2 2014, 15:13:35) [MSC v.1500 32 bit (Intel)] on win32
>>> import xray
>>> xray.__version__
'0.4.0'
```

Running the tests provided with xray generates the output below.

I understand the test errors probably result from the way in which Windows manages files, but the failures may be more significant.

```
$ nosetests xray

SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS..C:\Python27\Anaconda\lib\site-packages\scipy\io\netcdf.py:287: RuntimeWarning: Cannot close a netcdf_file opened with mmap=True, when netcdf_variables or arrays referring to its data still exist. All data arrays obtained from such files refer directly to data on disk, and must be copied before the file can be cleanly closed. (See netcdf_file docstring for more information on mmap.)
  ), category=RuntimeWarning)
.........C:\Python27\Anaconda\lib\site-packages\xray\core\utils.py:53: FutureWarning: numpy equal will not check object identity in the future. The comparison did not return the same result as suggested by the identity (`is`)) and will change.
  return ((arr1 == arr2) | (pd.isnull(arr1) & pd.isnull(arr2))).all()
.....EE.EE.E.EEE.E.EE.SSSSSSSSSSSSSSSS...S.SS.SSSSS.....SSS.............F...................S...................F....S......................F........................................S.............F............................F....................................................F.................................
======================================================================
ERROR: test_load_data (xray.test.test_backends.ScipyOnDiskDataTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 92, in test_load_data
    pass
  File "C:\Python27\Anaconda\lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 87, in assert_loads
    self.assertDatasetAllClose(expected, actual)
  File "C:\Python27\Anaconda\lib\contextlib.py", line 35, in __exit__
    self.gen.throw(type, value, traceback)
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 600, in roundtrip
    yield ds
  File "C:\Python27\Anaconda\lib\contextlib.py", line 35, in __exit__
    self.gen.throw(type, value, traceback)
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 273, in create_tmp_file
    os.remove(path)
WindowsError: [Error 32] The process cannot access the file because it is being used by another process: 'c:\\users\\breisfel\\appdata\\local\\temp\\tmpnniync.nc'

======================================================================
ERROR: test_orthogonal_indexing (xray.test.test_backends.ScipyOnDiskDataTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 211, in test_orthogonal_indexing
    self.assertDatasetAllClose(expected, actual)
  File "C:\Python27\Anaconda\lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 600, in roundtrip
    yield ds
  File "C:\Python27\Anaconda\lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 273, in create_tmp_file
    os.remove(path)
WindowsError: [Error 32] The process cannot access the file because it is being used by another process: 'c:\\users\\breisfel\\appdata\\local\\temp\\tmpanlrts.nc'

======================================================================
ERROR: test_roundtrip_None_variable (xray.test.test_backends.ScipyOnDiskDataTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 108, in test_roundtrip_None_variable
    self.assertDatasetAllClose(expected, actual)
  File "C:\Python27\Anaconda\lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 600, in roundtrip
    yield ds
  File "C:\Python27\Anaconda\lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 273, in create_tmp_file
    os.remove(path)
WindowsError: [Error 32] The process cannot access the file because it is being used by another process: 'c:\\users\\breisfel\\appdata\\local\\temp\\tmpinhb9v.nc'

======================================================================
ERROR: test_roundtrip_coordinates (xray.test.test_backends.ScipyOnDiskDataTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 182, in test_roundtrip_coordinates
    self.assertDatasetIdentical(original, actual)
  File "C:\Python27\Anaconda\lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 600, in roundtrip
    yield ds
  File "C:\Python27\Anaconda\lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 273, in create_tmp_file
    os.remove(path)
WindowsError: [Error 32] The process cannot access the file because it is being used by another process: 'c:\\users\\breisfel\\appdata\\local\\temp\\tmpezxsf_.nc'

======================================================================
ERROR: test_roundtrip_example_1_netcdf (xray.test.test_backends.ScipyOnDiskDataTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 165, in test_roundtrip_example_1_netcdf
    self.assertDatasetEqual(expected, actual)
  File "C:\Python27\Anaconda\lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 600, in roundtrip
    yield ds
  File "C:\Python27\Anaconda\lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 273, in create_tmp_file
    os.remove(path)
WindowsError: [Error 32] The process cannot access the file because it is being used by another process: 'c:\\users\\breisfel\\appdata\\local\\temp\\tmpo5zwxy.nc'

======================================================================
ERROR: test_roundtrip_mask_and_scale (xray.test.test_backends.ScipyOnDiskDataTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 253, in test_roundtrip_mask_and_scale
    self.assertDatasetAllClose(encoded, actual)
  File "C:\Python27\Anaconda\lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 600, in roundtrip
    yield ds
  File "C:\Python27\Anaconda\lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 273, in create_tmp_file
    os.remove(path)
WindowsError: [Error 32] The process cannot access the file because it is being used by another process: 'c:\\users\\breisfel\\appdata\\local\\temp\\tmp5jczpa.nc'

======================================================================
ERROR: test_roundtrip_object_dtype (xray.test.test_backends.ScipyOnDiskDataTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 137, in test_roundtrip_object_dtype
    self.assertDatasetIdentical(expected, actual)
  File "C:\Python27\Anaconda\lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 600, in roundtrip
    yield ds
  File "C:\Python27\Anaconda\lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 273, in create_tmp_file
    os.remove(path)
WindowsError: [Error 32] The process cannot access the file because it is being used by another process: 'c:\\users\\breisfel\\appdata\\local\\temp\\tmpji2z1n.nc'

======================================================================
ERROR: test_roundtrip_string_data (xray.test.test_backends.ScipyOnDiskDataTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 144, in test_roundtrip_string_data
    self.assertDatasetIdentical(expected, actual)
  File "C:\Python27\Anaconda\lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 600, in roundtrip
    yield ds
  File "C:\Python27\Anaconda\lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 273, in create_tmp_file
    os.remove(path)
WindowsError: [Error 32] The process cannot access the file because it is being used by another process: 'c:\\users\\breisfel\\appdata\\local\\temp\\tmp7sj2iu.nc'

======================================================================
ERROR: test_roundtrip_test_data (xray.test.test_backends.ScipyOnDiskDataTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 73, in test_roundtrip_test_data
    self.assertDatasetAllClose(expected, actual)
  File "C:\Python27\Anaconda\lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 600, in roundtrip
    yield ds
  File "C:\Python27\Anaconda\lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 273, in create_tmp_file
    os.remove(path)
WindowsError: [Error 32] The process cannot access the file because it is being used by another process: 'c:\\users\\breisfel\\appdata\\local\\temp\\tmpibvyaw.nc'

======================================================================
ERROR: test_write_store (xray.test.test_backends.ScipyOnDiskDataTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 68, in test_write_store
    self.assertDatasetAllClose(expected, actual)
  File "C:\Python27\Anaconda\lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 588, in create_store
    yield backends.ScipyDataStore(tmp_file, mode='w')
  File "C:\Python27\Anaconda\lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 273, in create_tmp_file
    os.remove(path)
WindowsError: [Error 32] The process cannot access the file because it is being used by another process: 'c:\\users\\breisfel\\appdata\\local\\temp\\tmpyy61n0.nc'

======================================================================
ERROR: test_zero_dimensional_variable (xray.test.test_backends.ScipyOnDiskDataTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 59, in test_zero_dimensional_variable
    self.assertDatasetAllClose(expected, actual)
  File "C:\Python27\Anaconda\lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 600, in roundtrip
    yield ds
  File "C:\Python27\Anaconda\lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 273, in create_tmp_file
    os.remove(path)
WindowsError: [Error 32] The process cannot access the file because it is being used by another process: 'c:\\users\\breisfel\\appdata\\local\\temp\\tmpfqemcm.nc'

======================================================================
FAIL: test_coords (xray.test.test_dataarray.TestDataArray)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_dataarray.py", line 408, in test_coords
    self.assertEquals(expected, actual)
AssertionError: 'Coordinates:\n  * x        (x) int64 -1 -2\n  * y        (y) int64 0 1 2' != 'Coordinates:\n  * x        (x) int32 -1 -2\n  * y        (y) int32 0 1 2'

======================================================================
FAIL: test_repr (xray.test.test_dataarray.TestDataArray)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_dataarray.py", line 34, in test_repr
    self.assertEqual(expected, repr(data_array))
AssertionError: "<xray.DataArray 'my_variable' (time: 2, x: 3)>\narray([[1, 2, 3],\n       [4, 5, 6]])\nCoordinates:\n    other    int64 0\n  * time     (time) int64 0 1\n  * x        (x) int64 0 1 2\nAttributes:\n    foo: bar" != "<xray.DataArray 'my_variable' (time: 2, x: 3)>\narray([[1, 2, 3],\n       [4, 5, 6]])\nCoordinates:\n    other    int32 0\n  * time     (time) int64 0 1\n  * x        (x) int64 0 1 2\nAttributes:\n    foo: bar"

======================================================================
FAIL: test_coords_properties (xray.test.test_dataset.TestDataset)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_dataset.py", line 335, in test_coords_properties
    self.assertEqual(expected, actual)
AssertionError: 'Coordinates:\n  * x        (x) int64 -1 -2\n  * y        (y) int64 0 1 2\n    a        (x) int64 4 5\n    b        int64 -10' != 'Coordinates:\n  * x        (x) int32 -1 -2\n  * y        (y) int32 0 1 2\n    a        (x) int32 4 5\n    b        int32 -10'

======================================================================
FAIL: test_to_and_from_dataframe (xray.test.test_dataset.TestDataset)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_dataset.py", line 1369, in test_to_and_from_dataframe
    assert expected.equals(actual), (expected, actual)
AssertionError: (   a    x
y        
0  1  bar,    a    x
y        
0  1  bar)

======================================================================
FAIL: test_converted_types (xray.test.test_variable.TestAsCompatibleData)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_variable.py", line 813, in test_converted_types
    self.assertEqual(np.dtype(int), actual.dtype)
AssertionError: dtype('int32') != dtype('int64')

======================================================================
FAIL: test_count (xray.test.test_variable.TestVariable)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_variable.py", line 758, in test_count
    self.assertEqual(actual.dtype, 'int64')
AssertionError: dtype('int32') != 'int64'

----------------------------------------------------------------------
Ran 368 tests in 5.705s

FAILED (SKIP=76, errors=11, failures=6)

```

Thank you for the prompt response. 

Here is the test output after installing netcdf4 via conda:

```
...........C:\Python27\Anaconda\lib\site-packages\xray\core\utils.py:53: FutureWarning: numpy equal will not check object identity in the future. The comparison did not return the same result as suggested by the identity (`is`)) and will change.
  return ((arr1 == arr2) | (pd.isnull(arr1) & pd.isnull(arr2))).all()
...E.....E..EE.................EE.S................SS.SSSS.SSSSSSES.....S.............................................F...................S...................F....S......................F........................................S.............F............................F....................................................F.................................
======================================================================
ERROR: test_write_store (xray.test.test_backends.NetCDF3ViaNetCDF4DataTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 68, in test_write_store
    self.assertDatasetAllClose(expected, actual)
  File "C:\Python27\Anaconda\lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 609, in create_store
    format='NETCDF3_CLASSIC')
  File "C:\Python27\Anaconda\lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 273, in create_tmp_file
    os.remove(path)
WindowsError: [Error 32] The process cannot access the file because it is being used by another process: 'c:\\users\\breisfel\\appdata\\local\\temp\\tmpcrovke.nc'

======================================================================
ERROR: test_dump_and_open_encodings (xray.test.test_backends.NetCDF4DataTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 539, in test_dump_and_open_encodings
    self.assertArrayEqual(ds.variables['time'], np.arange(10) + 4)
  File "C:\Python27\Anaconda\lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 273, in create_tmp_file
    os.remove(path)
WindowsError: [Error 32] The process cannot access the file because it is being used by another process: 'c:\\users\\breisfel\\appdata\\local\\temp\\tmpb6uvr4.nc'

======================================================================
ERROR: test_open_encodings (xray.test.test_backends.NetCDF4DataTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 518, in test_open_encodings
    self.assertDictEqual(actual_encoding, expected['time'].encoding)
  File "C:\Python27\Anaconda\lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 273, in create_tmp_file
    os.remove(path)
WindowsError: [Error 32] The process cannot access the file because it is being used by another process: 'c:\\users\\breisfel\\appdata\\local\\temp\\tmpw6k3xg.nc'

======================================================================
ERROR: test_open_group (xray.test.test_backends.NetCDF4DataTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 336, in test_open_group
    open_dataset(tmp_file, group='bar')
  File "C:\Python27\Anaconda\lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 273, in create_tmp_file
    os.remove(path)
WindowsError: [Error 32] The process cannot access the file because it is being used by another process: 'c:\\users\\breisfel\\appdata\\local\\temp\\tmpokud5r.nc'

======================================================================
ERROR: test_write_groups (xray.test.test_backends.NetCDF4DataTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 368, in test_write_groups
    self.assertDatasetIdentical(data2, actual2)
  File "C:\Python27\Anaconda\lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 273, in create_tmp_file
    os.remove(path)
WindowsError: [Error 32] The process cannot access the file because it is being used by another process: 'c:\\users\\breisfel\\appdata\\local\\temp\\tmpmk9aat.nc'

======================================================================
ERROR: test_write_store (xray.test.test_backends.NetCDF4DataTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 68, in test_write_store
    self.assertDatasetAllClose(expected, actual)
  File "C:\Python27\Anaconda\lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 281, in create_store
    yield backends.NetCDF4DataStore(tmp_file, mode='w')
  File "C:\Python27\Anaconda\lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 273, in create_tmp_file
    os.remove(path)
WindowsError: [Error 32] The process cannot access the file because it is being used by another process: 'c:\\users\\breisfel\\appdata\\local\\temp\\tmpioapvh.nc'

======================================================================
ERROR: test_write_store (xray.test.test_backends.ScipyOnDiskDataTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 68, in test_write_store
    self.assertDatasetAllClose(expected, actual)
  File "C:\Python27\Anaconda\lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 588, in create_store
    yield backends.ScipyDataStore(tmp_file, mode='w')
  File "C:\Python27\Anaconda\lib\contextlib.py", line 24, in __exit__
    self.gen.next()
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_backends.py", line 273, in create_tmp_file
    os.remove(path)
WindowsError: [Error 32] The process cannot access the file because it is being used by another process: 'c:\\users\\breisfel\\appdata\\local\\temp\\tmpspbwg3.nc'

======================================================================
FAIL: test_coords (xray.test.test_dataarray.TestDataArray)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_dataarray.py", line 408, in test_coords
    self.assertEquals(expected, actual)
AssertionError: 'Coordinates:\n  * x        (x) int64 -1 -2\n  * y        (y) int64 0 1 2' != 'Coordinates:\n  * x        (x) int32 -1 -2\n  * y        (y) int32 0 1 2'

======================================================================
FAIL: test_repr (xray.test.test_dataarray.TestDataArray)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_dataarray.py", line 34, in test_repr
    self.assertEqual(expected, repr(data_array))
AssertionError: "<xray.DataArray 'my_variable' (time: 2, x: 3)>\narray([[1, 2, 3],\n       [4, 5, 6]])\nCoordinates:\n    other    int64 0\n  * time     (time) int64 0 1\n  * x        (x) int64 0 1 2\nAttributes:\n    foo: bar" != "<xray.DataArray 'my_variable' (time: 2, x: 3)>\narray([[1, 2, 3],\n       [4, 5, 6]])\nCoordinates:\n    other    int32 0\n  * time     (time) int64 0 1\n  * x        (x) int64 0 1 2\nAttributes:\n    foo: bar"

======================================================================
FAIL: test_coords_properties (xray.test.test_dataset.TestDataset)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_dataset.py", line 335, in test_coords_properties
    self.assertEqual(expected, actual)
AssertionError: 'Coordinates:\n  * x        (x) int64 -1 -2\n  * y        (y) int64 0 1 2\n    a        (x) int64 4 5\n    b        int64 -10' != 'Coordinates:\n  * x        (x) int32 -1 -2\n  * y        (y) int32 0 1 2\n    a        (x) int32 4 5\n    b        int32 -10'

======================================================================
FAIL: test_to_and_from_dataframe (xray.test.test_dataset.TestDataset)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_dataset.py", line 1369, in test_to_and_from_dataframe
    assert expected.equals(actual), (expected, actual)
AssertionError: (   a    x
y        
0  1  bar,    a    x
y        
0  1  bar)

======================================================================
FAIL: test_converted_types (xray.test.test_variable.TestAsCompatibleData)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_variable.py", line 813, in test_converted_types
    self.assertEqual(np.dtype(int), actual.dtype)
AssertionError: dtype('int32') != dtype('int64')

======================================================================
FAIL: test_count (xray.test.test_variable.TestVariable)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27\Anaconda\lib\site-packages\xray\test\test_variable.py", line 758, in test_count
    self.assertEqual(actual.dtype, 'int64')
AssertionError: dtype('int32') != 'int64'

----------------------------------------------------------------------
Ran 368 tests in 7.074s

FAILED (SKIP=18, errors=7, failures=6)

```

Thank you for working to resolve these issues.

It is looking much better.

```
Python 2.7.8 |Anaconda 2.1.0 (32-bit)| (default, Jul  2 2014, 15:13:35) [MSC v.1500 32 bit (Intel)] on win32
>>> import xray
>>> xray.__version__
'0.4.0.dev'

$ nosetests xray
.............D:\Python27\Anaconda\lib\site-packages\xray\core\utils.py:54: FutureWarning: numpy equal will not check obj
ect identity in the future. The comparison did not return the same result as suggested by the identity (`is`)) and will
change.
  return ((arr1 == arr2) | (pd.isnull(arr1) & pd.isnull(arr2))).all()
.....SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS.................................SSSSSSSSSSSSSSSS...S.SS.SSSSS.....SS
S..................................S............................S.......................................................
........S..............................................F....................................................F...........
......................
======================================================================
FAIL: test_converted_types (xray.test.test_variable.TestAsCompatibleData)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "D:\Python27\Anaconda\lib\site-packages\xray\test\test_variable.py", line 813, in test_converted_types
    self.assertEqual(np.dtype(np.int64), actual.dtype)
AssertionError: dtype('int64') != dtype('int32')

======================================================================
FAIL: test_count (xray.test.test_variable.TestVariable)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "D:\Python27\Anaconda\lib\site-packages\xray\test\test_variable.py", line 758, in test_count
    self.assertEqual(actual.dtype, 'int64')
AssertionError: dtype('int32') != 'int64'

----------------------------------------------------------------------
Ran 395 tests in 4.786s

FAILED (SKIP=76, failures=2)

```

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -1,0 +1,35 @@
+# CI on Windows via appveyor
+# This file was based on Olivier Grisel's python-appveyor-demo
+
+environment:
+
+  matrix:
+    - PYTHON: "C:\\Python27-conda32"
+      PYTHON_VERSION: "2.7"
+      PYTHON_ARCH: "32"
+
+    - PYTHON: "C:\\Python34-conda64"
+      PYTHON_VERSION: "3.4"
+      PYTHON_ARCH: "64"
+
+install:
+  # Install miniconda Python
+  - "powershell ./ci/install_python.ps1"
+
+  # Prepend newly installed Python to the PATH of this build (this cannot be
+  # done from inside the powershell script as it would require to restart
+  # the parent CMD process).
+  - "SET PATH=%PYTHON%;%PYTHON%\\Scripts;%PATH%"
+
+  # Check that we have the expected version and architecture for Python
+  - "python --version"
+  - "python -c \"import struct; print(struct.calcsize('P') * 8)\""
+
+  # install xray and depenencies
+  - "conda install --yes --quiet pip nose numpy pandas scipy netCDF4"
+  - "python setup.py install"
+
+build: false
+
+test_script:
+  - "nosetests xray"

--- a/ci/install_python.ps1
+++ b/ci/install_python.ps1
@@ -1,0 +1,93 @@
+# Sample script to install Python and pip under Windows
+# Authors: Olivier Grisel, Jonathan Helmus and Kyle Kastner
+# License: CC0 1.0 Universal: http://creativecommons.org/publicdomain/zero/1.0/
+
+$MINICONDA_URL = "http://repo.continuum.io/miniconda/"
+$BASE_URL = "https://www.python.org/ftp/python/"
+
+
+function DownloadMiniconda ($python_version, $platform_suffix) {
+    $webclient = New-Object System.Net.WebClient
+    if ($python_version -eq "3.4") {
+        $filename = "Miniconda3-3.7.3-Windows-" + $platform_suffix + ".exe"
+    } else {
+        $filename = "Miniconda-3.7.3-Windows-" + $platform_suffix + ".exe"
+    }
+    $url = $MINICONDA_URL + $filename
+
+    $basedir = $pwd.Path + "\"
+    $filepath = $basedir + $filename
+    if (Test-Path $filename) {
+        Write-Host "Reusing" $filepath
+        return $filepath
+    }
+
+    # Download and retry up to 3 times in case of network transient errors.
+    Write-Host "Downloading" $filename "from" $url
+    $retry_attempts = 2
+    for($i=0; $i -lt $retry_attempts; $i++){
+        try {
+            $webclient.DownloadFile($url, $filepath)
+            break
+        }
+        Catch [Exception]{
+            Start-Sleep 1
+        }
+   }
+   if (Test-Path $filepath) {
+       Write-Host "File saved at" $filepath
+   } else {
+       # Retry once to get the error message if any at the last try
+       $webclient.DownloadFile($url, $filepath)
+   }
+   return $filepath
+}
+
+
+function InstallMiniconda ($python_version, $architecture, $python_home) {
+    Write-Host "Installing Python" $python_version "for" $architecture "bit architecture to" $python_home
+    if (Test-Path $python_home) {
+        Write-Host $python_home "already exists, skipping."
+        return $false
+    }
+    if ($architecture -eq "32") {
+        $platform_suffix = "x86"
+    } else {
+        $platform_suffix = "x86_64"
+    }
+    $filepath = DownloadMiniconda $python_version $platform_suffix
+    Write-Host "Installing" $filepath "to" $python_home
+    $install_log = $python_home + ".log"
+    $args = "/S /D=$python_home"
+    Write-Host $filepath $args
+    Start-Process -FilePath $filepath -ArgumentList $args -Wait -Passthru
+    if (Test-Path $python_home) {
+        Write-Host "Python $python_version ($architecture) installation complete"
+    } else {
+        Write-Host "Failed to install Python in $python_home"
+        Get-Content -Path $install_log
+        Exit 1
+    }
+}
+
+
+function InstallMinicondaPip ($python_home) {
+    $pip_path = $python_home + "\Scripts\pip.exe"
+    $conda_path = $python_home + "\Scripts\conda.exe"
+    if (-not(Test-Path $pip_path)) {
+        Write-Host "Installing pip..."
+        $args = "install --yes pip"
+        Write-Host $conda_path $args
+        Start-Process -FilePath "$conda_path" -ArgumentList $args -Wait -Passthru
+    } else {
+        Write-Host "pip already installed."
+    }
+}
+
+
+function main () {
+    InstallMiniconda $env:PYTHON_VERSION $env:PYTHON_ARCH $env:PYTHON
+    InstallMinicondaPip $env:PYTHON
+}
+
+main

--- a/doc/whats-new.rst
+++ b/doc/whats-new.rst
@@ -86,7 +86,7 @@ Bug fixes
   aggregation methods (:issue:`344`).
 - Dataset aggregations with ``keep_attrs=True`` now preserve attributes on
   data variables, not just the dataset itself.
-- Tests for xray now pass when run on Windows. DOUBLE CHECK THIS.
+- Tests for xray now pass when run on Windows (:issue:`360`).
 - Fixed a regression in v0.4 where saving to netCDF could fail with the error
   ``ValueError: could not automatically determine time units``.
 

--- a/xray/test/test_backends.py
+++ b/xray/test/test_backends.py
@@ -385,7 +385,7 @@ class NetCDF4DataTest(CFEncodedDataTest, TestCase):
 
             with open_dataset(tmp_file) as xray_dataset:
                 with create_tmp_file() as tmp_file2:
-                    xray_dataset.dump(tmp_file2)
+                    xray_dataset.to_netcdf(tmp_file2)
                     with nc4.Dataset(tmp_file2, 'r') as ds:
                         self.assertEqual(ds.variables['time'].getncattr('units'), units)
                         self.assertArrayEqual(ds.variables['time'], np.arange(10) + 4)
@@ -494,53 +494,6 @@ class NetCDF4DataTest(CFEncodedDataTest, TestCase):
         with self.roundtrip(data) as actual:
             self.assertDatasetIdentical(data, actual)
             self.assertEqual(actual['x'].dtype, np.dtype('S4'))
-
-    def test_open_encodings(self):
-        # Create a netCDF file with explicit time units
-        # and make sure it makes it into the encodings
-        # and survives a round trip
-        with create_tmp_file() as tmp_file:
-            with nc4.Dataset(tmp_file, 'w') as ds:
-                ds.createDimension('time', size=10)
-                ds.createVariable('time', np.int32, dimensions=('time',))
-                units = 'days since 1999-01-01'
-                ds.variables['time'].setncattr('units', units)
-                ds.variables['time'][:] = np.arange(10) + 4
-
-            expected = Dataset()
-
-            time = pd.date_range('1999-01-05', periods=10)
-            encoding = {'units': units, 'dtype': np.dtype('int32')}
-            expected['time'] = ('time', time, {}, encoding)
-
-            with open_dataset(tmp_file) as actual:
-                self.assertVariableEqual(actual['time'], expected['time'])
-                actual_encoding = dict((k, v) for k, v
-                                       in iteritems(actual['time'].encoding)
-                                       if k in expected['time'].encoding)
-                self.assertDictEqual(actual_encoding,
-                                     expected['time'].encoding)
-
-    def test_dump_and_open_encodings(self):
-        # Create a netCDF file with explicit time units
-        # and make sure it makes it into the encodings
-        # and survives a round trip
-        with create_tmp_file() as tmp_file:
-            with nc4.Dataset(tmp_file, 'w') as ds:
-                ds.createDimension('time', size=10)
-                ds.createVariable('time', np.int32, dimensions=('time',))
-                units = 'days since 1999-01-01'
-                ds.variables['time'].setncattr('units', units)
-                ds.variables['time'][:] = np.arange(10) + 4
-
-            xray_dataset = open_dataset(tmp_file)
-
-            with create_tmp_file() as tmp_file2:
-                xray_dataset.to_netcdf(tmp_file2)
-
-                with nc4.Dataset(tmp_file2, 'r') as ds:
-                    self.assertEqual(ds.variables['time'].getncattr('units'), units)
-                    self.assertArrayEqual(ds.variables['time'], np.arange(10) + 4)
 
     def test_coordinates_encoding(self):
         def equals_latlon(obj):

--- a/xray/test/test_variable.py
+++ b/xray/test/test_variable.py
@@ -755,7 +755,7 @@ class TestVariable(TestCase, VariableSubclassTestCases):
 
         actual = Variable(['x'], [True, False, True]).count()
         self.assertVariableIdentical(expected, actual)
-        self.assertEqual(actual.dtype, 'int64')
+        self.assertEqual(actual.dtype, int)
 
         expected = Variable(['x'], [2, 3])
         actual = Variable(['x', 'y'], [[1, 0, np.nan], [1, 1, 1]]).count('y')
@@ -810,7 +810,7 @@ class TestAsCompatibleData(TestCase):
             actual = _as_compatible_data(input_array)
             self.assertArrayEqual(np.asarray(input_array), actual)
             self.assertEqual(NumpyArrayAdapter, type(actual))
-            self.assertEqual(np.dtype(np.int64), actual.dtype)
+            self.assertEqual(np.asarray(input_array).dtype, actual.dtype)
 
     def test_masked_array(self):
         original = np.ma.MaskedArray(np.arange(5))

